scope3ai/api/tracer.py

```diff
@@ -71,5 +73,5 @@ def _link_trace(self, trace) -> None:
             self.traces.append(trace)
 
     def _unlink_trace(self, trace) -> None:
-        if trace in self.traces:
+        if trace in self.traces and not self.keep_tracers:
```

Comment:
Order could matter, `in` operator cost more than the `not X`. Can you just make it first and standalone as they are not directly related ?

```python
if not self.keep_tracers:
    return
```